plugins/inputs/docker_log/docker_logs.go

```diff
@@ -0,0 +1,308 @@
+	mu              sync.Mutex
+	containerList   map[string]io.ReadCloser
+}
+
+const (
+	defaultEndpoint = "unix:///var/run/docker.sock"
+	LOG_BYTES_MAX   = 1000
+)
+
+var (
+	containerStates = []string{"created", "restarting", "running", "removing", "paused", "exited", "dead"}
+)
+
+var sampleConfig = `
+  ## Docker Endpoint
+  ##   To use TCP, set endpoint = "tcp://[ip]:[port]"
+  ##   To use environment variables (ie, docker-machine), set endpoint = "ENV"
+  endpoint = "unix:///var/run/docker.sock"
+  ## Only collect metrics for these containers, collect all if empty
+  container_names = []
+  ## Containers to include and exclude. Globs accepted.
+  ## Note that an empty array for both will include all containers
+  container_name_include = []
+  container_name_exclude = []
+  ## Container states to include and exclude. Globs accepted.
+  ## When empty only containers in the "running" state will be captured.
+  # container_state_include = []
+  # container_state_exclude = []
+
+  ## docker labels to include and exclude as tags.  Globs accepted.
+  ## Note that an empty array for both will include all labels as tags
+  docker_label_include = []
+  docker_label_exclude = []
+
+  ## Optional TLS Config
+  # tls_ca = "/etc/telegraf/ca.pem"
+  # tls_cert = "/etc/telegraf/cert.pem"
+  # tls_key = "/etc/telegraf/key.pem"
+  ## Use TLS but skip chain & host verification
+  # insecure_skip_verify = false
+`
+
+func (d *DockerLogs) Description() string {
+	return "Plugin to get docker logs"
+}
+
+func (d *DockerLogs) SampleConfig() string {
+	return sampleConfig
+}
+
+func (d *DockerLogs) Gather(acc telegraf.Accumulator) error {
+	/*Check to see if any new containers have been created since last time*/
+	d.containerListUpdate(acc)
+	return nil
+}
+
+/*Following few functions have been inherited from telegraf docker input plugin*/
+func (d *DockerLogs) createContainerFilters() error {
+	// Backwards compatibility for deprecated `container_names` parameter.
+	if len(d.ContainerNames) > 0 {
+		d.ContainerInclude = append(d.ContainerInclude, d.ContainerNames...)
+	}
+
+	filter, err := filter.NewIncludeExcludeFilter(d.ContainerInclude, d.ContainerExclude)
+	if err != nil {
+		return err
+	}
+	d.containerFilter = filter
+	return nil
+}
+
+func (d *DockerLogs) createLabelFilters() error {
+	filter, err := filter.NewIncludeExcludeFilter(d.LabelInclude, d.LabelExclude)
+	if err != nil {
+		return err
+	}
+	d.labelFilter = filter
+	return nil
+}
+
+func (d *DockerLogs) createContainerStateFilters() error {
+	if len(d.ContainerStateInclude) == 0 && len(d.ContainerStateExclude) == 0 {
+		d.ContainerStateInclude = []string{"running"}
+	}
+	filter, err := filter.NewIncludeExcludeFilter(d.ContainerStateInclude, d.ContainerStateExclude)
+	if err != nil {
+		return err
+	}
+	d.stateFilter = filter
+	return nil
+}
+
+func (d *DockerLogs) addToContainerList(containerId string, logReader io.ReadCloser) error {
+	d.containerList[containerId] = logReader
```

Comment:
Should lock around the read/writes of this map

Reply:
Taken care of this!